on foo
   try
      -- code
   catch tError
      -- code
   finally
      -- code
   end try
   -- comment
end foo
